### ISSUE TITLE
Input component placeholders don't meet contrast requirements

### ISSUE THREAD

**Issue by robinmetral** [sumup-oss/circuit-ui#1816]
## Describe the bug

The placeholder color for the `Input` (and `CurrencyInput`, `SearchInput`, `TextArea`) and some legacy `Calendar*` inputs doesn't meet WCAG 2.1 AA color contrast requirements:

<img width="818" alt="The placeholder color has a contrast ratio of 2.85:1" src="https://user-images.githubusercontent.com/35560568/198024495-c8808b17-d858-4601-befe-b3e39f21c5e9.png">

## Expected behavior

All test should meet contrast requirements, including placeholder text.

Two options:

- make placeholder text darker to meet the requirements (risks being confused for input value)
- [stop](https://www.nngroup.com/articles/form-design-placeholders/) [using](https://www.smashingmagazine.com/2018/06/placeholder-attribute/) [placeholders](https://www.deque.com/blog/accessible-forms-the-problem-with-placeholders/)

## Specifications

* Version: latest
* Browser: all
* OS: all

## Additional context

> The minimum contrast success criterion (1.4.3) applies to text in the page, **_including placeholder text_**

–WCAG 2.1 [SC 1.4.3 Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) (emphasis mine)

**Comment by robinmetral** [comment]
Still relevant for obvious reasons (this is a WCAG violation)

**Comment by connor-baer** [comment]
Still relevant. We're revamping the input design at the moment and will review the use of placeholders as part of this effort.
